test/batch_test.rb

```diff
@@ -9,10 +9,13 @@ def test_batch
   end
 
   def test_nested_batch
-    GraphQL::Batch.batch do
-      assert_raises(GraphQL::Batch::NestedError) do
-        GraphQL::Batch.batch {}
+    product = GraphQL::Batch.batch do
+      promise = RecordLoader.for(Product).load(1)
+      GraphQL::Batch.batch do
+        RecordLoader.for(Product).load(1)
```

Comment:
You aren't testing anything with this.  Perhaps you could assert that it returns the same promise object as the one above.

Reply:
Done.